File: paper_artifacts/experiments/llr8/artifacts/sources/kimi27sglang-fortran-skills/tsvc_2_s1232__submitted.f90
subroutine tsvc_2_s1232_fp64(aa, bb, cc, len_2d, vlen, workspace, workspace_size) bind(C)
  use iso_c_binding
  implicit none
  integer(c_int64_t), value, intent(in) :: len_2d, vlen, workspace_size
  real(c_double), intent(inout) :: aa(len_2d, len_2d)
  real(c_double), intent(in) :: bb(len_2d, len_2d)
  real(c_double), intent(in) :: cc(len_2d, len_2d)
  integer(c_int8_t), intent(inout) :: workspace(workspace_size)
  integer(c_int64_t) :: i, j, jmax

  !$omp parallel do schedule(dynamic, 16)
  do i = 1, len_2d
    jmax = (i - 1) / vlen + 1
    !$omp simd
    do j = 1, jmax
      aa(j, i) = bb(j, i) + cc(j, i)
    end do
  end do
end subroutine tsvc_2_s1232_fp64

File: paper_artifacts/experiments/llr9/artifacts/sources/kimi27sglang-c-skills/tsvc_2_s1232__submitted.c
#include <stdint.h>
#include <omp.h>

void tsvc_2_s1232_fp64(double *restrict aa, double *restrict bb,
                       double *restrict cc, int64_t LEN_2D, int64_t VLEN,
                       uint8_t *workspace, int64_t workspace_bytes)
{
    (void)workspace;
    (void)workspace_bytes;

    #pragma omp parallel for schedule(guided)
    for (int64_t i = 0; i < LEN_2D; i++) {
        const int64_t jmax = i / VLEN;
        for (int64_t j = 0; j <= jmax; j++) {
            aa[i * LEN_2D + j] = bb[i * LEN_2D + j] + cc[i * LEN_2D + j];
        }
    }
}
